plan/plan.go

```diff
@@ -458,21 +476,25 @@ func pullUpUnaryExprs(n *Node) error {
 }
 
 // Pull children of nested merge joins up to the top level.
-func pullUpMergeJoins(n *Node) error {
-	if n.Type != MergeJoin {
-		return nil
-	}
-	newChildren := []*Node{}
-	queue := []*Node{n}
-	for len(queue) > 0 {
-		node := queue[0]
-		queue = queue[1:]
-		if node.Type == MergeJoin {
-			queue = append(queue, node.Children...)
-			continue
+// Also appends the descending flag to the merge join node.
```

Comment:
I think it would be cleaner to structure this as a separate pushdown function:

```
pushDownDescending(descending bool) func(*Node) error {
if n.Type == MergeJoin || n.Type == Scan {
  n.Args = append(n.Args, descending)
}
return nil
}
```

then compose that with the pushdowns. One issue that will introduce is it'll force you to put descending after the variadic section of the scan node arg list. To get around that I wonder if we should just add a "descending" flag to the plan.Node structure and convey it through that means instead.

Reply:
I think with this you don't need the changes in the two existing pushdowns